src/mlpack/methods/ann/layer/subview.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file subview.hpp
+ * @author Haritha Nair
+ *
+ * Definition of the Subview class, which modifies the input as necessary.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SUBVIEW_HPP
+#define MLPACK_METHODS_ANN_LAYER_SUBVIEW_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/activation_functions/identity_function.hpp>
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * Implementation of the subview layer. The subview layer modifies the input to
+ * a submatrix of required size.
+ *
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class Subview
+{
+ public:
+  /**
+   * Create the Subview layer object using the specified range of input to accept.
+   *
+   * @param begin Start index.
+   * @param end End index.
+   */
+  Subview(const arma::uword begin = 0, const arma::uword end = 0):
+    begin(begin),
+    end(end)
+  {
+    /* Nothing to do here */
+  }
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(InputType&& input, OutputType&& output)
+  {
+    // Check if input has been selected as required.
```

Comment:
I was wondering if we could use one of armadillo's advanced constructors to reuse the memory, something like:

```c++
output = arma::Mat<ElemType>(input.memptr() + begin, 1, end - begin, false, false);
```

Reply:
Ofcourse, I'll edit it. Just one doubt, shouldn't the `n_rows` parameter be `(end-begin)` and `n_cols` be `input.n_cols`, considering we are passing matrices and not just vectors?

Reply:
You are right, just thought about vectors.